src/gui/general_menu.py

```diff
@@ -55,92 +39,13 @@ def button_setup(self):
         # create buttons
         for title in self.options:
             rect = generic_button_rect
-            button = Button(title, self.font, rect, self.rect.topleft)
+            button = Button(title, rect, self.font)
             self.buttons.append(button)
             generic_button_rect = rect.move(0, button_height + space)
 
-    # events
-    def event_loop(self):
-        self.mouse_hover()
-        for event in pygame.event.get():
-            if event.type == pygame.QUIT:
-                self.quit_game()
-
-            self.click(event)
-            self.handle_events(event)
-
-    def handle_events(self, event):
-        pass
-    
-
-    def get_hovered_button(self):
-        for button in self.buttons:
-            if button.mouse_hover():
-                return button
-        return None
-
-    def click(self, event):
-        if event.type == pygame.MOUSEBUTTONDOWN and mouse_buttons()[0]:
-            self.pressed_button = self.get_hovered_button()
-            if self.pressed_button:
-                self.pressed_button.start_press_animation()
-
-        if event.type == pygame.MOUSEBUTTONUP:
-            if self.pressed_button:
-                self.pressed_button.start_release_animation()
-
-                if self.pressed_button.mouse_hover():
-                    self.button_action(self.pressed_button.text)
-                    pygame.mouse.set_cursor(pygame.SYSTEM_CURSOR_ARROW)
-
-                self.pressed_button = None
-
-    def mouse_hover(self):
-        for button in self.buttons:
-            if button.hover_active:
-                pygame.mouse.set_cursor(pygame.SYSTEM_CURSOR_HAND)
-                return
-        pygame.mouse.set_cursor(pygame.SYSTEM_CURSOR_ARROW)
-
     def button_action(self, text):
         if text == 'Play':
+            pygame.mouse.set_cursor(pygame.SYSTEM_CURSOR_ARROW)
```

Comment:
Yup, basically, quitting out of any menu back to gameplay when hovering around a button would keep the hand cursor active instead of the default arrow